modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationAfterAccountDisablingTestCase.java

```diff
@@ -0,0 +1,445 @@
+        if (ArrayUtils.isNotEmpty(claimValues)) {
+            accountDisabledClaimValue = claimValues[0].getValue();
+        }
+        Assert.assertTrue(Boolean.parseBoolean(accountDisabledClaimValue), "User account didn't disabled");
+    }
+
+    @Test(description = "Check whether access token is revoked after disabling the account",
+            dependsOnMethods = "disableUserAccount")
+    private void introspectAccessTokenOfDisabledAccount() throws Exception {
+
+        Set<String> appKeys = applications.keySet();
+        for (String appName : appKeys) {
+            TokenIntrospectionResponse revokedTokenIntrospectionResponse =
+                    introspectAccessToken(accessTokens.get(appName),
+                            privilegedAccessTokens.get(appName));
+            Assert.assertTrue(revokedTokenIntrospectionResponse.indicatesSuccess(),
+                    "Failed to receive a success response.");
+            Assert.assertFalse(revokedTokenIntrospectionResponse.toSuccessResponse().isActive(),
+                    "Introspection response of a revoked access token is successful.");
+        }
+    }
+
+    private void createServiceProviderApplication() throws Exception {
+
+        OAuthConsumerAppDTO appDTO = new OAuthConsumerAppDTO();
+        appDTO.setApplicationName(OAUTH_APPLICATION_NAME_1);
+        appDTO.setCallbackUrl(OAuth2Constant.CALLBACK_URL);
+        appDTO.setOAuthVersion(OAuth2Constant.OAUTH_VERSION_2);
+        appDTO.setTokenType(tokenType);
+        appDTO.setGrantTypes("authorization_code implicit password client_credentials refresh_token "
+                + "urn:ietf:params:oauth:grant-type:saml2-bearer iwa:ntlm");
+        OAuthConsumerAppDTO oAuthConsumerAppDTO = createApplication(appDTO, SERVICE_PROVIDER_1_NAME);
+        applications.put(SERVICE_PROVIDER_1_NAME, oAuthConsumerAppDTO);
+
+        OAuthConsumerAppDTO appDTO2 = new OAuthConsumerAppDTO();
+        appDTO2.setApplicationName(OAUTH_APPLICATION_NAME_2);
+        appDTO2.setCallbackUrl(OAuth2Constant.CALLBACK_URL);
+        appDTO2.setOAuthVersion(OAuth2Constant.OAUTH_VERSION_2);
+        appDTO2.setTokenType(tokenType);
+        appDTO2.setGrantTypes("authorization_code implicit password client_credentials refresh_token "
+                + "urn:ietf:params:oauth:grant-type:saml2-bearer iwa:ntlm");
+        OAuthConsumerAppDTO oAuthConsumerAppDTO2 = createApplication(appDTO2, SERVICE_PROVIDER_2_NAME);
+        applications.put(SERVICE_PROVIDER_2_NAME, oAuthConsumerAppDTO2);
+    }
+
+    private void addNewTestUserWithRole() throws Exception {
+
+        remoteUSMServiceClient.addUser(TEST_USER_USERNAME, TEST_USER_PASSWORD, null, null,
+                PROFILE_NAME, false);
+        userMgtClient = new UserManagementClient(backendURL, sessionCookie);
+        userMgtClient.addRole(TEST_ROLE, new String[] {TEST_USER_USERNAME},
+                new String[] {"/permission/admin/login"}, false);
```

Comment:
New role is not mandatory. Double check!

Reply:
Fixed in https://github.com/wso2/product-is/pull/14727/commits/4882f9bbdb620ceaaf5e04a15be585a6c1676420